pkg/gadgets/internal/socketenricher/bpf/sockets-map.bpf.c

```diff
@@ -63,6 +64,7 @@ insert_socket_from_iter(struct sock *sock, struct task_struct *task)
 	struct sockets_value socket_value = {0,};
 	// use given task
 	socket_value.pid_tgid = ((u64)task->tgid) << 32 | task->pid;
+	socket_value.uid_gid = ((u64)task->cred->gid.val) << 32 | task->cred->uid.val;
```

Comment:
I think it should be `s/cred/real_cred`:

Here task is not the current task but another task that might be running VFS calls. The VFS code might temporary substitute `task->cred` by other creds during overlayfs copyup. In this case, we want the real creds of the process, not the creds temporarily substituted by VFS overlayfs copyup.

https://github.com/inspektor-gadget/inspektor-gadget/commit/690f74bd11a91fbfda6ba8f8cdbe22f8daee8176#diff-6baf489ee945d6f76e3e02d7755a2a890b548321c57f414fc3af36b47a0ab33aR101-R106

Reply:
makes sense. Updated!